draft-ietf-tls-tls13.md

```diff
@@ -2430,8 +2432,8 @@ The "extension_data" field of this extension contains a
 
 client_shares
 : A list of offered KeyShareEntry values in descending order of client preference.
-  This vector MUST NOT be empty. Clients not providing a KeyShare MUST instead
-  omit this extension from the ClientHello.
+  Servers MUST NOT negotiate any connection if this vector is empty and instead
+  MUST respond with a HelloRetryRequest (see {{hello-retry-request}}).
```

Comment:
See also issue #277. If the (EC)DHE suite is higher priority, it MUST NOT try and find another lower priority suite. That just risks downgrading when the client was trying to request a server group selection via retry. If there's a higher priority suite than the (EC)DHE one(s), then we don't even get to the point of caring about this extension. This could be stated in detail.